### PR TITLE
Sync container VM power state when refreshing [specific ci=Group1-Docker-Commands]

### PR DESCRIPTION
**Pull request by dougm**
Issue #4872

### CODE REVIEW

**Comment by mdubya66** [comment]
Nice

**Comment by cgtexmex** [comment]
When you issue a remove the persona will call and check state -- that call results in an update to the container on the PL.  The Refresh() method is called which initiates a property collector call and ultimately an update to the `base` container -- are we not propagating the updated status to the `container` object?  Just curious if we need this new call or do we just need to fix what we have?

File: lib/portlayer/exec/container.go [hunk 1]
```diff
@@ -546,7 +546,11 @@ func (c *Container) Remove(ctx context.Context, sess *session.Session) error {
 
 	// check state first
 	if c.state == StateRunning {
-		return RemovePowerError{fmt.Errorf("Container is powered on")}
+		// double check to workaround issue 4872
+		state, _ := c.vm.PowerState(ctx)
```
Comment (anchal-agrawal) [inline lib/portlayer/exec/container.go]: We should add a `// #nosec: Errors unhandled.` above this for the gas tool.

**Comment by dougm** [comment]
@cgtexmex I'll look at changing Refresh() to update the power state, it does not do so currently.

**Comment by hmahmood** [comment]
@cgtexmex @dougm `Refresh()` does quite a lot; maybe just add another helper function to just refresh the state?

**Comment by sflxn** [comment]
@cgtexmex I saw something similar in the issue I merged in today where the property refresh was not getting the correct power state.  My suspicion was vSphere correcting it's internal FSM while we were refreshing properties.  In my issue, VIC's internal state tracking was right and VC knew the VM was powered off but property refresh got incorrect state.

**Comment by dougm** [comment]
@hmahmood can you re-review?

@sflxn that's not the problem here, see recent comments in the linked issue.

**Comment by cgtexmex** [comment]
@dougm you should change your commit message...we are no longer double checking but just propagating the state during a refresh...

**Comment by sflxn** [comment]
We should add [specific ci=Group1-Docker-Commands] to test this against through the docker commands to see if everything is fine first.

**Comment by sflxn** [comment]
lgtm.  but you need Caglar or George to lgtm.

**Pull request merged**